compiler/rustc_driver_impl/src/lib.rs

```diff
@@ -362,8 +364,6 @@ fn run_compiler(
                 result
             })?;
 
-            drop(gctxt);
```

Comment:
This was explicitly added so that `GlobalCtxt` could be freed before invoking the linker to reduce peak memory usage. Has this been intentionally changed?

Reply:
I guess the one I added in https://github.com/rust-lang/rust/pull/56732 was a couple of lines below. I wonder when that went away. It seems like that could easily slip by perf as I assume it doesn't track total memory usage regressions.

Reply:
They are tracked by perf, but only really checked by looking at the over-time graphs

Reply:
Are you sure? I though it only tracked the `rustc` process.

Reply:
https://perf.rust-lang.org/index.html?start=&end=&kind=percentfromfirst&stat=max-rss shows the max rss changes of all tests over time

Reply:
How would this affect memory usage? AFAICT, `gctxt` is a `QueryResult<QueryContext>`. `QueryContext` is just a wrapper for a `&'tcx GlobalCtxt`. Also, `Queries::ongoing_codegen` and `Queries::linker` both call `self.global_ctxt()`, so the query is repeated there anyway.

Reply:
This PR just happened to remind me of an earlier state of rustc_interface where `queries.global_ctxt()` returned an owned copy which was explictly dropped before the linker was run. The current state of the compiler looks good.